pkgs/development/tools/backblaze-b2/default.nix

```diff
@@ -75,6 +76,15 @@ python3Packages.buildPythonApplication rec {
       --zsh <(${python3Packages.argcomplete}/bin/register-python-argcomplete backblaze-b2)
   '';
 
+  passthru.tests.version = (testers.testVersion {
+    package = backblaze-b2;
+    command = "backblaze-b2 version --short";
+  }).overrideAttrs (old: {
+    # workaround the error: Permission denied: '/homeless-shelter'
+    # backblaze-b2 fails to create a 'b2' directory under the XDG config path
+    HOME = "/tmp";
```

Comment:
this works fine,also can be `$(mktemp -d)`

Reply:
Please do this. `TMPDIR` is not always /tmp.

Reply:
good point

nix will use whatever value is in `TMPDIR` or `/tmp`
https://github.com/NixOS/nix/blob/2.19.2/src/libstore/build/local-derivation-goal.cc#L2099

Reply:
I changed this to `$(mktemp -d)` as suggested.